agent/checks/check.go

```diff
@@ -541,6 +542,7 @@ func (c *CheckH2PING) check() {
 	}
 	ctx, cancel := context.WithTimeout(context.Background(), c.Timeout)
 	defer cancel()
+	defer clientConn.Shutdown(ctx)
```

Comment:
In order to guarantee a graceful shutdown, it might not be a bad idea to have a new context for the shutdown with the same length of time as the timeout- instead of using the remainder of the timeout time, but that could double the time of each check so I am curious to know any thoughts.

Reply:
We could create a new context with a fraction of the original timeout (maybe half?) to be consistent about waiting for the shutdown.